File: backend/src/modules/products/dto/create-product.dto.ts
import { IsNotEmpty, IsNumber, IsOptional, IsString, IsUrl, Min } from 'class-validator';

export class CreateProductDto {
    @IsNotEmpty()
    @IsString()
    title: string;

    @IsOptional()
    @IsString()
    description?: string;

    @IsNotEmpty()
    @IsNumber()
    @Min(100) // Mínimo 1 real (100 centavos)
    priceCents: number;

    @IsNotEmpty()
    @IsUrl()
    redirectUrl: string;

    @IsOptional()
    @IsUrl()
    imageUrl?: string;
}

File: backend/Capicash.API/DTOs/CreateProductDto.cs
using System.ComponentModel.DataAnnotations;

using System.Text.Json.Serialization;

namespace Capicash.API.DTOs;

public class CreateProductDto
{
    [Required]
    [JsonPropertyName("title")]
    public string Title { get; set; } = string.Empty;

    [JsonPropertyName("description")]
    public string? Description { get; set; }

    [Required]
    [Range(1, 10000000)] // Min 1 centavo, Max 100k
    [JsonPropertyName("priceCents")] 
    public int PriceInCents { get; set; }

    [JsonPropertyName("redirectUrl")]
    public string? RedirectUrl { get; set; }

    [JsonPropertyName("contentUrl")]
    public string? ContentUrl { get; set; }

    [JsonPropertyName("imageUrl")]
    public string? ImageUrl { get; set; }
}
